contracts/multisig/src/contract/execute.rs

```diff
@@ -214,16 +212,16 @@ pub fn authorize_callers(
 
 pub fn unauthorize_callers(
     deps: DepsMut,
-    contracts: Vec<Addr>,
+    callers: Vec<Addr>,
 ) -> error_stack::Result<Response, ContractError> {
-    let callers_and_chains = contracts
+    let callers_and_chains = callers
         .into_iter()
-        .map(|contract| {
-            remove_prover_from_registry(deps.storage, contract.clone())
+        .filter_map(|contract| {
+            remove_prover(deps.storage, &contract)
                 .map(|cc| cc.map(|chain| (contract, chain)))
-                .change_context(ContractError::InvalidCaller)
+                .change_context(ContractError::InvalidProver)
```

Comment:
this error seems off too, because from current context we can't tell if the error was caused by an invalid prover. So, maybe use something like `ContractError:Storage` or `ProverRemovalFailed`?